replications/internal/queue_management.go

```diff
@@ -143,3 +148,91 @@ func (qm *durableQueueManager) CurrentQueueSizes(ids []platform.ID) (map[platfor
 
 	return sizes, nil
 }
+
+// StartReplicationQueues updates the durableQueueManager.replicationQueues map, fully removing any partially deleted
+// queues (present on disk, but not tracked in sqlite), opening all current queues, and logging info for each.
+func (qm *durableQueueManager) StartReplicationQueues(trackedReplications map[platform.ID]int64) error {
+	errOccurred := false
+
+	for id, size := range trackedReplications {
+		// Re-initialize a queue struct for each replication stream from sqlite
+		queue, err := durablequeue.NewQueue(
+			filepath.Join(qm.queuePath, id.String()),
+			size,
+			durablequeue.DefaultSegmentSize,
+			&durablequeue.SharedCount{},
+			durablequeue.MaxWritesPending,
+			func(bytes []byte) error {
+				return nil
+			},
+		)
+
+		if err != nil {
+			qm.logger.Error("failed to initialize replication stream durable queue", zap.Error(err))
+			errOccurred = true
+		}
+
+		// Open and map the queue struct to its replication ID
+		if err := queue.Open(); err != nil {
+			qm.logger.Error("failed to open replication stream durable queue", zap.Error(err), zap.String("id", id.String()))
+			errOccurred = true
+		} else {
+			qm.replicationQueues[id] = queue
+			qm.logger.Info("Opened replication stream", zap.String("id", id.String()), zap.String("path", queue.Dir()))
+		}
+	}
+
+	if errOccurred {
+		return fmt.Errorf("startup tasks for replications durable queue management failed, see server logs for details")
```

Comment:
This could use `errors.New` instead of `fmt.Errorf`, and be lifted to a top-level `var` for reuse